modules/local_storage/src/StockLocationStorage.php

```diff
@@ -34,4 +34,48 @@ public function loadEnabled(PurchasableEntityInterface $entity) {
     return $enabled_locations;
   }
 
+  public function loadInContext(Context $context, PurchasableEntityInterface $entity) {
+
+    $store = $context->getStore();
+    // Make sure we have the availability field for the location.
+    if ($store->hasField('field_available_stock_locations')) {
+      // Get the available locations.
+      $locations = $store->field_available_stock_locations->getValue();
+      // Load them.
+      $store_locations = array();
+      foreach ($locations as $location) {
+        $store_locations[$location['target_id']] = $location['target_id'];
+      }
+      $store_locations = $this->loadMultiple($store_locations);
+      // Remove if not enabled.
+      foreach ($store_locations as $id => $location) {
+        if (!$location->isActive()) {
+          unset($store_locations[$id]);
+        }
+      }
+      return $store_locations;
+    }
+    else {
+      return $this->loadEnabled($entity);
+    }
+  }
+
+
+  public function getTransactionLocation(Context $context, PurchasableEntityInterface $entity) {
+    $store = $context->getStore();
+    // Make sure we have the availability field for the location.
+    if ($store->hasField('field_stock_allocation_location')) {
```

Comment:
To support multiple stores you must add **field_stock_allocation_location** to your store